Sources/CountDownLatch.swift

```diff
@@ -44,14 +50,14 @@ public class CountDownLatch {
     }
 
     public func countDown() {
-        OSAtomicDecrement32(&currentCount)
+        currentCount -= 1
+        
         if currentCount == 0 {
             waitSemaphore.signal()
         }
 
         if currentCount < 0 {
             print("Count Down decreased more times than expected.")
         }
```

Comment:
This should solve the threading issue, though we may still see some inconsistencies.
Do we need to execute these as an atomic operation to make sure another thread does not change currentCount while the other one is reading, e.g. 
```
initialCount = 1
t1 countDown() L53
t2 countDown() L53
t1 countDown() L55 true
t2 countDown() L55 false